Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1710,6 +1858,25 @@ namespace AzToolsFramework
                 PrefabDom entityDomAfter(&domToAddDuplicatedEntitiesUnder.GetAllocator());
                 entityDomAfter.Parse(newEntityDomString.toUtf8().constData());
 
+                EntityAlias parentEntityAlias;
+                if (auto componentsIter = entityDomAfter.FindMember(PrefabDomUtils::ComponentsName);
+                    componentsIter != entityDomAfter.MemberEnd())
+                {
+                    for (auto componentIter = componentsIter->value.MemberBegin(); componentIter != componentIter->value.MemberEnd() && parentEntityAlias.empty();
+                         ++componentIter)
+                    {
+                        if (auto parentEntityIter = componentIter->value.FindMember("Parent Entity");
+                            parentEntityIter != componentIter->value.MemberEnd())
+                        {
+                            parentEntityAlias = parentEntityIter->value.GetString();
+                            break;
+                        }
+                    }
+                }
+
+                // Insert our entity into its parent's sort order
+                AddNewEntityToSortOrder(commonOwningInstance, domToAddDuplicatedEntitiesUnder, parentEntityAlias, newEntityAlias);
```

Comment:
Since you are doing this in DuplicateNestedEntitiesInInstance() function, should we be also doing the same in DuplicateNestedInstancesInInstance function? The later will take care of duplicating prefabs. Or is that solved using a different mechanism?

Reply:
Maybe? It looks like `DuplicateNestedInstancesInInstance` is only ever called from `DuplicateNestedEntitiesInInstance` and it adds the aliases to the map, so I think it's working right now because `DuplicateNestedEntitiesInInstance` does the fixup.

Reply:
DuplicateNestedInstancesInInstance is called after DuplicateNestedEntitiesInInstance is called. Not from it.